redfish-core/include/utils/chassis_utils.hpp

```diff
@@ -61,5 +61,258 @@ void getValidChassisID(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     BMCWEB_LOG_DEBUG << "checkChassisId exit";
 }
 
+template <typename Callback>
+inline void
+    checkAssemblyInterface(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                           const std::string& chassisPath,
+                           std::vector<std::string>& assemblies,
+                           Callback&& callback)
+{
+    crow::connections::systemBus->async_method_call(
+        [aResp, chassisPath, assemblies, callback{std::move(callback)}](
+            const boost::system::error_code ec,
+            const std::vector<std::pair<
+                std::string,
+                std::vector<std::pair<std::string, std::vector<std::string>>>>>&
+                subtree) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "D-Bus response error on GetSubTree " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            if (subtree.size() == 0)
+            {
+                BMCWEB_LOG_DEBUG << "No object paths found";
+                return;
+            }
+            std::vector<std::string> updatedAssemblyList;
+            for (const auto& [objectPath, serviceName] : subtree)
+            {
+                // This list will store common paths between assemblies fetched
+                // from association json and assemblies which are actually
+                // implemeted. This is to handle the case in which there is
+                // entry in association json but implementation of interface for
+                // that particular assembly is missing.
+                auto it =
+                    std::find(assemblies.begin(), assemblies.end(), objectPath);
+                if (it != assemblies.end())
+                {
+                    updatedAssemblyList.emplace(updatedAssemblyList.end(), *it);
+                }
+            }
+
+            if (updatedAssemblyList.size() != 0)
+            {
+                // sorting is required to facilitate patch as the array does not
+                // store and data which can be mapped back to Dbus path of
+                // assembly.
```

Comment:
This comment could be irrelevant in our case since we are not doing a patch; but we do need to sort to find the position of the requested assembly from this list...